### ISSUE TITLE
Request: Time-editable Dashboards for Collaboration

### ISSUE THREAD

**Issue by joethreepwood** [PostHog/posthog#8460]
## Is your feature request related to a problem?

This request was suggested by a customer in a conversation for an upcoming case study. He described it as something he experiences daily friction with, which limits his ability to collaborate and which he'd love to see resolved. Quote: "_This would save my life if you guys did it, honestly. You have no idea._"

The short version is that the individual is a Product Manager who is responsible for several internal growth metrics. He often shares information with other teams who do not have or need full PostHog access. He specifically called out people in Sales, Marketing and Leadership as examples of this. These teams are stakeholders in the decisions he makes and he often finds himself sharing data with them to explain upcoming product changes or experiments. 

He does this primarily through shared public dashboards, though he does occasionally share screenshots too. However, the behaviour he frequently sees is that users want to explore the dashboard data in a little more detail - _especially_ in terms of the date/time range. 

They are unable to make these changes however, so they then reach out to him to request more detail - which means he has to create additional insights, update dashboards, etc. Sometimes the data they are looking for is already on the dashboard, but he believes the date shown in an insight on such a dashboard is not sufficiently clear. 

Quote: "_I have my dashboard for my team and we have our dashboard for the projects that we're working on. So, I try to save everything on the dashboard and I just share the link. And I think that's one of the challenges is...the way that the dashboard shows up is that, just read sometimes, it can be overwhelming for them. I can understand it, maybe because I'm use it, but I have a more data driven concept in my mind. And I have an engineer background as well, even though I'm a product manager. But what I have been seeing is people getting lost a lot when I share the dashboard link...One of the things people are asking for a lot is they want to change, for example, the period of time of one graphic. They want to change that and play around and see. But I only want to see last month, two months ago and I don't want them to break the dashboard!_"

## Describe the solution you'd like

There are probably multiple solutions to this. The one we discussed was the idea of a 'protected mode' for dashboards, where users are able to play with the available data in a dashboard but cannot save the changes permanently. 

## Describe alternatives you've considered

Another possible, more focused solution, would be the ability to change the date/time scale on dashboards (without saving changes to the insight) when data is shared. 

This probably wants to be behind some sort of toggle, as not all insights will want to have such 'explorability'. 

## Additional context

Tagging for @paolodamico in light of https://github.com/PostHog/product-internal/pull/261#discussion_r799748761

**Comment by joethreepwood** [comment]
Missed off this one quote: 

"Because that's one of the challenge. Because every time for example, I give the permission for people to go there and change, they screw up the dashboard. So I have to go there, change, take a screenshot and send to them. So if I can give just the ability for them to play around here without breaking anything, that's going to be awesome."

**Comment by neilkakkar** [comment]
Some related old discussion: https://github.com/PostHog/posthog/issues/5358

I like the new context here around working with *public dashboards* and how non-editable shouldn't imply non-interactable. (Fair summary of problem components, @joethreepwood ?)

**Comment by joethreepwood** [comment]
> non-editable shouldn't imply non-interactable.

This is exactly correct. Thank you!

**Comment by clarkus** [comment]
I agree, but I think there are limits. If the user is in an editing context, I think they should be able to save their changes. That said, I'd like to see some parameters editable in a non-permanent way. Things that help users put the current chart into context - time ranges, time units, etc. These are baked into the insight chart view already, so they're available when building and when viewing. 

<img width="1213" alt="Screen Shot 2022-02-07 at 10 51 05 AM" src="https://user-images.githubusercontent.com/254612/152852575-7abaf0a4-06be-4c9a-8f4c-594ceea00b73.png">

If the user wants to use the insight or the dashboard as a starting point for their own exploration, that's what duplication is for. You're creating a new insight that matches all the parameters of the source insight. It solves the same job, but it preserves our permission boundaries and ensures your insights don't change without your consent. Thoughts?

For dashboards, I think there might be cases for different [dashboard types](https://github.com/PostHog/product-internal/pull/261#issuecomment-1030332253).

**Comment by paolodamico** [comment]
Thanks for sharing this @joethreepwood, it's super helpful context. I think it's worth discussing the high-level problems we want to solve with dashboards first (https://github.com/PostHog/product-internal/pull/267) and then figure out how we can have a good UX for this. We're trying to solve so many things with dashboards, this could feel confusing or convoluted, that's the reason it might be worth taking a bird's eye view here first.

**Comment by paolodamico** [comment]
Related to https://github.com/PostHog/product-internal/pull/267, if we continue/improve allowing users to play around with dashboards we should consider treating each date change as another "dashboard viewed"/"dashboard analyzed", it's the same behavior as insights and intuitively you would expect that changing these filters would lead to new discoveries. Thoughts @marcushyett-ph ?

**Comment by joethreepwood** [comment]
Bit of a resurrection, but I also just experienced some friction with this issue too when trying to look at one of @andyvan-ph's queries for SEO articles. He's locked it down, so I can't look at that query over a longer time. Result? I have the PITA of rebuilding a duplicate query.

**Comment by marcushyett-ph** [comment]
I believe this is something @mariusandra is looking into right now.

**Comment by clarkus** [comment]
This should be possible right now on cloud - from an insight **read** view, you can change the date without requiring an edit or save action. Likewise you can change the time grouping units. Insight read views are missing a read-only translation of the query structure right now. This is available on insight cards / dashboard items though. 

<img width="377" alt="Screen Shot 2022-04-19 at 8 05 58 AM" src="https://user-images.githubusercontent.com/254612/164035359-f21dbaca-f4e5-479e-ba1e-1e1fa2e0d35f.png">

**Comment by joethreepwood** [comment]
I may be misunderstanding @clarkus but I'm not currently able to change the date without having edit access? See below. 

<img width="628" alt="Screenshot 2022-04-19 at 16 24 29" src="https://user-images.githubusercontent.com/84011561/164039503-05632e6c-da29-4080-8a6d-c4f70da228a9.png">

**Comment by clarkus** [comment]
Ah - I forgot I was on my admin account. Yeah I think that these fields should be available to anyone who can view the insight. You're not actually editing the thing, so it doesn't violate any permissions or boundaries. Said another way, if you can see it then you should be able to change the date parameters. @mariusandra any ideas here?

**Comment by Twixes** [comment]
The above should be fixed by #9467, which allows editing insight configuration in view mode in a read-only way.
